SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -139,81 +139,366 @@ struct ActorSystem::RepStateUpdateHelper
 			{
-				ComponentRemoved(Delta.EntityId, Change.ComponentId);
+				GActorSystem->ComponentUpdated(Delta.EntityId, Change.ComponentId, Change.Update);
 			}
-			for (const AuthorityChange& Change : Delta.AuthorityGained)
+			for (const ComponentChange& Change : Delta.ComponentsRefreshed)
 			{
-				AuthorityGained(Delta.EntityId, Change.ComponentSetId);
+				GActorSystem->ApplyComponentAdd(Delta.EntityId, Change.ComponentId, Change.CompleteUpdate.Data);
+				GActorSystem->ComponentAdded(Delta.EntityId, Change.ComponentId, Change.CompleteUpdate.Data);
 			}
-			for (const AuthorityChange& Change : Delta.AuthorityLostTemporarily)
+			for (const ComponentChange& Change : Delta.ComponentsRemoved)
 			{
-				AuthorityGained(Delta.EntityId, Change.ComponentSetId);
+				GActorSystem->ComponentRemoved(Delta.EntityId, Change.ComponentId);
 			}
 			break;
 		}
-		case EntityDelta::ADD:
-			PopulateDataStore(Delta.EntityId);
-			EntityAdded(Delta.EntityId);
-			break;
-		case EntityDelta::REMOVE:
-			EntityRemoved(Delta.EntityId);
-			ActorDataStore.Remove(Delta.EntityId);
-			break;
-		case EntityDelta::TEMPORARILY_REMOVED:
-			EntityRemoved(Delta.EntityId);
-			ActorDataStore.Remove(Delta.EntityId);
-			PopulateDataStore(Delta.EntityId);
-			EntityAdded(Delta.EntityId);
-			break;
 		default:
 			break;
 		}
 	}
+}
+
+ActorHandler::ActorHandler(USpatialNetDriver& InNetDriver, ViewCoordinator& Coordinator)
+	: ActorSubView(Coordinator.CreateSubView(SpatialConstants::ACTOR_TAG_COMPONENT_ID, &FMainActorSubviewSetup::IsActorEntity,
+											 FMainActorSubviewSetup::GetCallbacks(Coordinator)))
+	, ServerAuthoritySubView(
+		  Coordinator.CreateSubView(SpatialConstants::ACTOR_AUTH_TAG_COMPONENT_ID,
+									GetRoleFilterPredicate(ActorSubView, &FAuthoritySubviewSetup::IsAuthorityActorEntity, &InNetDriver),
+									FAuthoritySubviewSetup::GetCallbacks(Coordinator)))
+	, AutonomousProxySubView(Coordinator.CreateSubView(
+		  SpatialConstants::ACTOR_AUTH_TAG_COMPONENT_ID,
+		  GetRoleFilterPredicate(ActorSubView, &FAutonomousSubviewSetup::IsAutonomousActorEntity, &ServerAuthoritySubView, &InNetDriver),
+		  FAutonomousSubviewSetup::GetCallbacks(Coordinator)))
+	, SimulatedProxySubView(
+		  Coordinator.CreateSubView(SpatialConstants::ACTOR_TAG_COMPONENT_ID,
+									GetRoleFilterPredicate(ActorSubView, &FSimulatedSubviewSetup::IsSimulatedActorEntity,
+														   &ServerAuthoritySubView, &AutonomousProxySubView, &InNetDriver),
+									FSimulatedSubviewSetup::GetCallbacks(Coordinator)))
+	, NetDriver(InNetDriver)
+{
+}
+
+static const TCHAR* ToCharArray(const ENetMode NetMode)
+{
+	switch (NetMode)
+	{
+	case NM_Standalone:
+		return TEXT("Standalone");
+	case NM_DedicatedServer:
+		return TEXT("DedicatedServer");
+	case NM_ListenServer:
+		return TEXT("ListenServer");
+	case NM_Client:
+		return TEXT("Client");
+	default:
+		checkNoEntry();
+		return TEXT("Invalid");
+	}
+}
+
+void ActorHandler::Advance(ActorSystem& InActorSystem)
```

Comment:
This is the meat of this PR: different subviews are processed in a well-defined order, and entities can be removed from processing by modifying the subview (ownership completeness is the use case for this)